README.md

```diff
@@ -231,3 +231,218 @@ var_dump($taichungService->getAction1()->do()->getMeaningData());
+
+Similar to the JSON RPC single request approach, you can use the `setBatchRpcQuery()` method in the `Action` object to pass the desired `method`, `params`, and `id` as an array to perform batch RPC connections.
+
+Once the `setBatchRpcQuery()` method is enabled, the HTTP verb of the `Action` will automatically change to `POST`.
+
+If all batch `RPC` responses are successfully parsed, you can handle your `RPC` responses using the `doneHandler`.
+
+```php
+require './vendor/autoload.php';
+use SDPMlab\Anser\Service\Action;
+use Psr\Http\Message\ResponseInterface;
+use SDPMlab\Anser\Service\ServiceList;
+
+$id = 1;
+$action = (new Action(
+    'http://myRpcServer.com',
+    "POST",
+    "/"
+))
+->setBatchRpcQuery([
+    ["/myRpcMethod", [1,2], $id],
+    ["/myRpcMethod", [1,2], $id],
+    ["/myRpcMethod", [1,2], $id]
+]); 
+->doneHandler(static function(
+    ResponseInterface $response,
+    Action $runtimeAction
+) {
+    $returnData = [];
+    $rpcResponse = $action->getRpcResponse();
+    $resultArray = $action->getRpcResult();
+    $idArray     = $action->getId();
+    for ($i = 0; $i<3; $i++ ){
+        $returnData[] = [
+            "id" => $idArray[$i],
+            "result" => $resultArray[$i],
+        ]
+    }
+
+    $runtimeAction->setMeaningData($returnData);
+});
+
+$data = $action->do()->getMeaningData();
+var_dump($data);
+```
+
+### HTTP JSON RPC Error Handling - RPC Response Error
+
+You can configure a `failHandler` callback function to instruct the `Action` on how to handle the logic when encountering RPC errors, such as using `isRpcError()`,RPC error format ,[reference to JSON-RPC format](https://www.jsonrpc.org/ "Reference to JSON-RPC format")。
+
+You can use `getRpcResponse()` to obtain all RPC responses, and the `Action` has categorized their response statuses into `success` and `error`.
+
+If you only need to retrieve `RPC` responses of a specific status, you can use `getSuccessRpc()` and `getErrorRpc()`, which return arrays containing these responses. Additionally, you can use the data retrieval methods shown in the provided code to handle the data more precisely.
+
+```php
+<?php
+require './vendor/autoload.php';
+
+use \SDPMlab\Anser\Service\Action;
+use \Psr\Http\Message\ResponseInterface;
+use \SDPMlab\Anser\Exception\ActionException;
+
+$action = (new Action(
+    "http://myRpcServer.com",
+    "POST",
+    "/"
+))
+->setRpcQuery("/errorMethod", [1,2], $id)
+->doneHandler(function (
+    ResponseInterface $response,
+    Action $runtimeAction
+) {
+    $rpcResponse = $action->getRpcResponse();
+    $result = $action->getRpcResult()[0];
+    $id     = $action->getId()[0];
+    $runtimeAction->setMeaningData([
+        "id" => $id,
+        "result" => $result
+    ]);
+})->failHandler(function (
+    ActionException $e
+) {
+    if ($e->isRpcError()) {
+        $rpcResponse = $e->getRpcResponse()
+        $errorResArr = $e->getErrorRpc();
+        $successResArr = $e->getSuccessRpc();
```

Comment:
I think the return result of both`$e->getErrorRpc();` and `$e->getSuccessRpc();` could be a Key/Value array with `QueryID` as the key.